src/main/java/za/co/neslotech/spring/trainspringbootinsight/service/CrudService.java

```diff
@@ -0,0 +1,17 @@
+package za.co.neslotech.spring.trainspringbootinsight.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface CrudService<T, I> {
+
+    T update(T entity);
+    List<T> findAll();
+    Optional<T> findById(I id);
```

Comment:
So you can return the type of T. Instead of the optional, you can then allow the implementation to handle the not found case of the optional.